### PR TITLE
Project Extent | Save drawn points in the database

### PR DESCRIPTION
**Pull request by mddilley**
Closes part of https://github.com/cityofaustin/atd-data-tech/issues/5466

This PR extends the draw UI to save the drawn points in the `project_extent_geojson` field of projects so they can be selected and removed again after the initial save. The save button is a placeholder (just like the toggle switch) that is tied to a save function. The next PR for this feature will add the ability to see the drawn points in the Project Summary map alongside the streets and points and some refinements (https://github.com/cityofaustin/atd-data-tech/issues/5749).

### Demo
savePoints

### CODE REVIEW

**Review by amenity — APPROVED** [review]
Looks and works great, @mddilley !

File: moped-editor/src/utils/mapDrawHelpers.js [hunk 1]
```diff
@@ -91,17 +96,67 @@ export function getFeatureStyle({ feature, state }) {
...
+ */
+const findDifferenceByFeatureProperty = (featureProperty, arrayOne, arrayTwo) =>
+  arrayOne.filter(
+    arrayOneFeature =>
+      !arrayTwo.some(
+        arrayTwoFeature =>
+          arrayOneFeature.properties[featureProperty] ===
+          arrayTwoFeature.properties[featureProperty]
+      )
+  );
```
Comment (sergiogcx) [inline moped-editor/src/utils/mapDrawHelpers.js:124]: This is brilliant! TIL about the `some` method 💯

File: moped-editor/src/utils/mapDrawHelpers.js [hunk 2]
```diff
@@ -111,6 +166,79 @@ export function useMapDrawTools() {
...
+          "PROJECT_EXTENT_ID",
+          drawnFeatures,
+          featuresAlreadyInDrawMap
+        );
+
+        ref.addFeatures(featuresToAdd);
+      }
+    },
+    [featureCollection]
+  );
```
Comment (sergiogcx) [inline moped-editor/src/utils/mapDrawHelpers.js:191]: This is pretty intense code, the useCallback returns a function that adds only new features to a component?

File: moped-editor/src/utils/mapDrawHelpers.js [hunk 3]
```diff
@@ -185,7 +331,10 @@ export function useMapDrawTools() {
   const renderMapDrawTools = () => (
     <>
       <Editor
-        ref={mapEditorRef}
+        ref={ref => {
+          initializeExistingDrawFeatures(ref);
+          mapEditorRef.current = ref;
+        }}
```
Comment (sergiogcx) [inline moped-editor/src/utils/mapDrawHelpers.js:337]: This is advanced react, thanks for explaining how refs work!

**Review by sergiogcx — APPROVED** [review]
It works flawlessly! 💯

**Review by mateoclarke — APPROVED** [review]
🚢

**Pull request merged**